src/components/UriViewer/LogViewer.ts

```diff
@@ -0,0 +1,224 @@
+import _ from 'lodash/fp';
+import { useCallback, useEffect, useState } from 'react';
+import { div, h, span } from 'react-hyperscript-helpers';
+import { SimpleTabBar } from 'src/components/tabBars';
+import { isAzureUri } from 'src/components/UriViewer/uri-viewer-utils';
+import { Ajax } from 'src/libs/ajax';
+import { useCancellation, useOnMount, withDisplayName } from 'src/libs/react-utils';
+import { newTabLinkProps } from 'src/libs/utils';
+
+import { ButtonOutline } from '../common';
+import { centeredSpinner, icon } from '../icons';
+import Modal from '../Modal';
+import { InfoBox } from '../PopupTrigger';
+/**
+ * Information needed to preview a log file.
+ * @member logUri - The URI of the log file. Must be a valid Azure blob URI. No Sas token should be appended: a fresh one will be obtained.
+ * @member logTitle - The title of the log. Displayed to the user as a tab title.
+ * @member logKey - A unique key for this particular log.
+ * @member logFilename - The filename of this particular log. Does not need to be unique.
+ */
+export type LogInfo = {
+  logUri: string;
+  logTitle: string;
+  logKey: string;
+  logFilename: string;
+};
+
+/**
+ * Props for the LogViewer component.
+ * @member modalTitle - Should represent the group of logs being displayed. E.g. "Task Logs"
+ * @member logs - An array of LogInfo objects.
+ */
+export type LogViewerProps = {
+  modalTitle: string;
+  logs: LogInfo[];
+  onDismiss: () => void;
+};
+
+/**
+ * Props for the SimpleTab component. See src/components/tabBars.js
+ */
+type SimpleTabProps = {
+  key: string;
+  title: string;
+  width: number;
+};
+
+type FetchedLogData = {
+  textContent: string | undefined; // The literal characters of the log file.
+  downloadUri: string | undefined; // The URI to use for downloading the log file. May or may not have token appended, depending on if the file is public or private.
+};
+
+const modalMaxWidth = 1100;
+const tabMaxWidth = modalMaxWidth / 4 - 20;
+
+export const LogViewer = _.flow(withDisplayName('LogViewer'))(({ modalTitle, logs, onDismiss }: LogViewerProps) => {
```

Comment:
```suggestion
export const LogViewer = ({ modalTitle, logs, onDismiss }: LogViewerProps) => {
```

`withDisplayName` is generally used to update the display name of a component that's been created by wrapping another component in one or more layers of higher order components. In this case, since there's no such wrapping, it's not needed.